kornia/augmentation/_2d/intensity/motion_blur.py

```diff
@@ -81,17 +81,23 @@ def __init__(
         self._param_generator = rg.MotionBlurGenerator(kernel_size, angle, direction)
         self.flags = dict(border_type=BorderType.get(border_type), resample=Resample.get(resample))
 
-    def generate_parameters(self, batch_shape: torch.Size) -> Dict[str, Tensor]:
+    def generate_parameters(self, batch_shape: torch.Size):
         params = super().generate_parameters(batch_shape)
-        params["idx"] = torch.randint(batch_shape[0], (1,)).item()
+        params["idx"] = tensor([0]) if batch_shape[0] == 0 else torch.randint(batch_shape[0], (1,))
```

Comment:
BTW, just remembered, what about the `same_on_batch` option here? If we want to produce the exact behaviour across the whole batch, is this still work?

Reply:
@shijianjian it will work. This index is used to select the blur kernel size since we use conv2d which requires same kernel size for all samples in the batch.